### PR TITLE
fabrics: create a persistent controller using unique subsystem nqn

### PR DESCRIPTION
**Pull request by martin-gpy**
One should ideally create a persistent discovery controller
using the unique subsystem nqn if supported, rather than
the well-known subsystem nqn.

Signed-off-by: Martin George <marting@netapp.com>

### CODE REVIEW

File: fabrics.c [hunk 1]
```diff
@@ -505,6 +505,31 @@ static int discover_from_conf_file(nvme_root_t r, nvme_host_t h,
...
+							trsvcid);
+					if (!c)
+						goto next;
+					nvme_ctrl_set_discovery_ctrl(c, true);
+					errno = 0;
+					ret = nvmf_add_ctrl(h, c, &cfg);
+					if (ret)
+						goto next;
+				}
+			}
```
Comment (igaw) [inline fabrics.c:532]: I think this should go into a function. The code below looks almost identically.

**Review by igaw — CHANGES_REQUESTED** [review]

File: fabrics.c (hunk 1 above)
Comment (martin-gpy) [reply fabrics.c:532]: > I think this should go into a function. The code below looks almost identically.

You mean a common helper function? 

I did think about that initially, but then one would have to pass all the nvme_create_ctrl() & nvmf_add_ctrl() params to that helper function, which would make it look quite ugly and unwieldy. So dropped that idea...

Comment (igaw) [reply fabrics.c:532]: I don't. Having almost identically code but does the same thing is a nightmare to maintain.

**Comment by igaw** [comment]
Merged #1490 instead. Closing this one.

**Pull request closed without merging**